heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -62,6 +74,13 @@ public Slave(SlaveLooper slaveLooper,
     this.streamInCommunicator = streamInCommunicator;
     this.streamOutCommunicator = streamOutCommunicator;
     this.inControlQueue = inControlQueue;
+    this.metricsOutCommunicator = metricsOutCommunicator;
+    this.serializer =
+        SerializeDeSerializeHelper.getSerializer(helper.getTopologyContext().getTopologyConfig());
+
+    // TODO (nlu): Create the state with corresponding type according to config
+    instanceState = new HashMapState<>();
```

Comment:
This empty state is just for initialization. If the instance is instructed to restore from a state, they will receive a `RestoreInstanceStateRequest` request, and thus `handleRestoreInstanceStateRequest ` will take care of loading the state